Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/SendSearchRequestToDrivers/Handle/Internal/DriverPool.hs

```diff
@@ -373,31 +381,45 @@ fetchScore ::
+fetchScore merchantOpCityId driverActualDistanceList driverIds intelligentPoolConfig driverPoolCfg cancellationScoreRelatedConfig mbWeeklyLeaderBoardConfig factor =
+  HM.fromList <$> case (factor, mbWeeklyLeaderBoardConfig) of
+    (AcceptanceRatio, _) | intelligentPoolConfig.acceptanceRatioWeightage /= 0 -> do
       acceptanceRatios <- getRatios (getLatestAcceptanceRatio merchantOpCityId) driverIds
       getScoreWithWeight (intelligentPoolConfig.acceptanceRatioWeightage) acceptanceRatios
-    CancellationRatio | intelligentPoolConfig.cancellationRatioWeightage /= 0 -> do
+    (CancellationRatio, _) | intelligentPoolConfig.cancellationRatioWeightage /= 0 -> do
       cancellationRatios <-
         getRatios (getLatestCancellationRatio cancellationScoreRelatedConfig merchantOpCityId) driverIds
       getScoreWithWeight (intelligentPoolConfig.cancellationRatioWeightage) cancellationRatios
-    AvailableTime | intelligentPoolConfig.availabilityTimeWeightage /= 0 -> do
+    (AvailableTime, _) | intelligentPoolConfig.availabilityTimeWeightage /= 0 -> do
       let maxAvailbaleTime = fromIntegral $ intelligentPoolConfig.availabilityTimeWindowOption.period * convertPeriodTypeToSeconds intelligentPoolConfig.availabilityTimeWindowOption.periodType
       driversAvailableTimeRatio <- map (second ((/ maxAvailbaleTime) . sum . catMaybes)) <$> getRatios (getCurrentWindowAvailability merchantOpCityId) driverIds
       getScoreWithWeight (intelligentPoolConfig.availabilityTimeWeightage) driversAvailableTimeRatio
-    DriverSpeed | intelligentPoolConfig.driverSpeedWeightage /= 0 -> do
+    (DriverSpeed, _) | intelligentPoolConfig.driverSpeedWeightage /= 0 -> do
       averageSpeeds <- getRatios (getDriverAverageSpeed merchantOpCityId . cast) driverIds
       getSpeedScore (intelligentPoolConfig.driverSpeedWeightage) averageSpeeds
-    ActualPickupDistance | intelligentPoolConfig.actualPickupDistanceWeightage /= 0 -> do
+    (ActualPickupDistance, _) | intelligentPoolConfig.actualPickupDistanceWeightage /= 0 -> do
       pure $ map (bimap (.getId) ((* (fromIntegral intelligentPoolConfig.actualPickupDistanceWeightage)) . fromIntegral . flip div (fromMaybe (Meters 1) (driverPoolCfg.actualDistanceThreshold)))) driverActualDistanceList
+    (RideFrequency, Just weeklyleaderBoardconfig) | intelligentPoolConfig.numRidesWeightage /= 0 -> do
+      currDate <- RideEndInt.getCurrentDate <$> getCurrentTime
+      let (_, currDayIndex) = DTCO.sundayStartWeek currDate
+      let weekStartDate = addDays (fromIntegral (- currDayIndex)) currDate
+      let weekEndDate = addDays (fromIntegral (6 - currDayIndex)) currDate
+      let scoreKey = RideEndInt.makeCachedWeeklyDriverLeaderBoardKey driverPoolCfg.merchantId weekStartDate weekEndDate
+      logDebug $ "The weekly score key in RideFrequency Pooling is " <> show scoreKey
+      scoreMap :: (HM.Map Text Double) <- Redis.safeGet scoreKey <&> maybe HM.empty HM.fromList
```

Comment:
this key will have data only for top n drivers from the leaderboard not the score of all drivers, also it will not work from second batch onwards because these keys are not set using `crossAppRedis`